OpenRA.Mods.Common/Activities/Move/Nudge.cs

```diff
@@ -0,0 +1,81 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class Nudge : Activity
+	{
+		readonly Mobile mobile;
+		readonly Aircraft aircraft;
+		readonly Actor nudger;
+		readonly bool queued;
+		public Nudge(Actor self, Actor nudger, bool queued = false)
+		{
+			this.nudger = nudger;
+			this.queued = queued;
+
+			mobile = self.TraitOrDefault<Mobile>();
+			aircraft = self.TraitOrDefault<Aircraft>();
+		}
+
+		protected override void OnFirstRun(Actor self)
+		{
+			if (mobile != null)
+			{
+				if (mobile.IsTraitDisabled || mobile.IsTraitPaused || mobile.IsImmovable)
+					return;
+
+				var cell = mobile.GetAdjacentCell(nudger.Location);
+				if (cell != null)
+				{
+					if (queued)
+						QueueChild(mobile.MoveTo(cell.Value, 0, targetLineColor: mobile.Info.TargetLineColor));
+					else
+						Queue(mobile.MoveTo(cell.Value, 0, targetLineColor: mobile.Info.TargetLineColor));
+				}
+			}
+			else if (aircraft != null)
+			{
+				if (aircraft.IsTraitDisabled || aircraft.IsTraitPaused || aircraft.RequireForceMove)
+					return;
+
+				// Disable nudging if the aircraft is outside the map
+				if (!self.World.Map.Contains(self.Location))
+					return;
+
+				var offset = new WVec(0, -self.World.SharedRandom.Next(512, 2048), 0)
+					.Rotate(WRot.FromFacing(self.World.SharedRandom.Next(256)));
+				var target = Target.FromPos(self.CenterPosition + offset);
+
+				if (queued)
+					QueueChild(new Fly(self, target, targetLineColor: aircraft.Info.TargetLineColor));
+				else
+					Queue(new Fly(self, target, targetLineColor: aircraft.Info.TargetLineColor));
```

Comment:
What is this for? Fwiw, this not how order.Queued is used.

Reply:
Tbf it doesn't matter now that I've overriden `TargetLineNodes`. The problem I was having is that target line nodes are calculated from the parent activity, and by using Queue I would make sure this activity ends (is not parent)